torchtext/transforms.py

```diff
@@ -189,3 +193,350 @@ def forward(self, input: Any) -> Any:
         :rtype: Union[List[Union[str, int]], List[List[Union[str, int]]]]
         """
         return F.truncate(input, self.max_seq_len)
+
+
+class GPT2BPETokenizer(Module):
+    """
+    Transform for GPT-2 BPE Tokenizer. 
+
+    Reimplements openai GPT-2 BPE in TorchScript. Original openai implementation 
+    https://github.com/openai/gpt-2/blob/master/src/encoder.py
+
+    :param bpe_encoder: A dict mapping from bpe tokens(e.g sub-words) to
+            bpe token ids. For example: {"ingly": 4420}
+    :type bpe_encoder: Dict[str, int]
+    :param bpe_merge_ranks: A dict mapping a pair of bpe tokens to rank.
+            Torchscript doesn't support tuple as dict key type, concat
+            a pair of string with SEPERATOR (e.g \u0001) as workaround.
+            For example: {"ing\u0001ly": 210}
+    :type bpe_merge_ranks: Dict[str, int]
+    :param byte_encoder: A dict mapping a byte to an unicode character.
+    :type byte_encoder: Dict[int, str]
+    """
+    SEPERATOR: str = "\u0001"
+    LSTRIP_PATTERN: str = "\u0120"
+
+    def __init__(
+        self,
+        bpe_encoder: Dict[str, int],
+        bpe_merge_ranks: Dict[str, int],
+        byte_encoder: Dict[int, str],
+    ):
```

Comment:
+1 to @Nayef211  comments below. We should probably just accept encoder_json_path and vocab_bpe_path inside the `__init__` and keep all other aspects of implementation as private. Since we are not yet have design for tokenizer. From Transform perspective, it would probably be just enough to have forward method as public. Also nit: we should probably give more intuitive names to these variables.

Reply:
> +1 to @Nayef211 comments below. We should probably just accept encoder_json_path and vocab_bpe_path inside the `__init__` and keep all other aspects of implementation as private. Since we do not yet have design for tokenizer. From Transform perspective, it would probably be just enough to have forward method as public. Also nit: we should probably give more intuitive names to these variables.

Note: In general, rather than taking a path object, taking a file-like object or the data with proper structure makes the interface more flexible as you can test with dummy data without need of dumping data to disk. IMO making a file path a public interface would this class a bit restricted, especially the actual data structure this class requires is just a dictionary. If I want to hide the loading mechanism I would use a uion of 1file-like object` (not a path object) and the proper structure, `Dict[std, int]]`. (but it's my spontaneous thinking so, there should be a better approach.)

Reply:
Sure, I think the suggestions make sense. We could keep the interface same. The loading can be handled by the client code, given it is just a plain python dictionary.

Reply:
Actually I just realized that bpe_merge_ranks is a special dictionary that we construct internally (using the separator) and we cannot expect this to be constructed by the client code. It is very much specific to our implementation as also noted by @abhinavarora in the [comment](https://github.com/pytorch/text/pull/1462#discussion_r769210112). We are better off sticking to the standard path or file-like objects pointing to the encoder_json_path  and vocab_bpe_path 
 instead of accepting dictionary. This way we can handle all the the complexity internally and the interface would accept the standard files generated by other frameworks as noted in the above comment.

Reply:
okay, that makes sense. I would still argue that loading mechanism can stay outside of the class so that class is only responsible for the processing logic but that would introduce one more public API so I am okay either way.

I wonder if it would be valuable in long term to implement a dictionary in C++ that takes tuple as key in TorchScript-compatible manner.

Reply:
Marking this as resolved for now!